### PR TITLE
Make the Selenium suite run headless, wait on conditions, and report failures

### PR DESCRIPTION
**Pull request by crcrewso**
## Summary

The GUI suite could not run headless, and could not be trusted when it did.
This replaces the harness, removes the timing assumptions, and gives a failed
browser test something to show for itself.

**Headless, natively.** `PyVirtualDisplay`/Xvfb is gone. Firefox and Chromium
both have native headless modes, and Selenium >= 4.10 resolves its own drivers
through Selenium Manager — so no display server, and no driver download step to
get wrong. `SELENIUM_BROWSER` and `SELENIUM_HEADLESS` pick the combination.

**Waits instead of sleeps.** The suite mixed implicit and explicit waits —
documented as undefined behaviour — and padded the difference with
`time.sleep()`. Every wait now names its condition. The races this exposed were
already present:

| Test | What it actually waited for |
|---|---|
| `test_submit_autosave` | nothing — it passed by luck |
| `login()` | "any page", satisfied before login completes |
| `load_admin()` | the same vacuous wait |
| the calendar test | the whole page, not the picker binding |
| `test_load_autosave` | the inputs to exist, not to be populated |

Removing the sleeps didn't cause these; it revealed them. `test_submit_autosave`'s
race exists byte-identically on `develop` today — it just doesn't manifest there.

**A screenshot when a GUI test fails.** `conftest.py` captures the browser on
failure into `selenium-screenshots/`; CI uploads it as an artifact. A GUI failure
that only reproduces on CI otherwise leaves no record of what the page looked
like. Nothing is written when tests pass.

**Skipped by default, selectable on purpose.** A plain `pytest` run no longer
needs a browser — GUI tests are skipped unless you pass `--run-selenium` or
select them with `-m selenium`. That's what makes the suite usable without a
browser installed, and what lets CI run them as their own job.

**CI.** Selenium runs on Linux against Firefox *and* Chromium, and on Windows
against Chromium. Other jobs install from the committed lockfile with a pinned
toolchain, and test in-memory sqlite alongside the file-backed one — they
exercise different code, and the in-memory path doubles as coverage of the
settings fallback itself.

Also here: three assertions stopped encoding today's date format; stub tests
marking the Selenium coverage that doesn't exist yet; `runtests.sh` no longer
bypasses the pytest config; `make test-<engine>` and `make test-integration`;
`setup.cfg` removed (its ruff config already lives in `pyproject.toml`);
`pyproject` license metadata corrected from MIT to Apache-2.0; developer-guide
sections on test ordering, remaining flakiness, finding unreached views, and
coverage gaps; the 4.0.0 schema diagram regenerated.

This is test infrastructure and CI only — no application code changes.

> _AI assistance (Claude / Claude Code) was used throughout this branch and to
> draft this summary. The disclosure checkboxes below are left for the PR author
> to complete._

## Related Issues

None.

## Target Branch

`develop`.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature ( adds functionality, always assumed to be breaking until tested by maintainers)
- [ ] Breaking change (fix or addition that changes existing behaviour)
- [ ] Documentation only

## Target Branch

_Confirm which branch this targets (e.g. master)._

## AI Usage Disclosure

- [ ] No AI tools were used in this PR.
- [ ] AI tools were used:
  - **Tools / where used:** _e.g. GitHub Copilot for boilerplate_
  - [ ] I've reviewed every AI-generated line as if I wrote it myself, and confirm: no PHI shared, no license-incompatible content (Apache-2.0).

---

### Coder Tasks

- [ ] Rebased / merged `develop` branch; no merge conflicts.
- [ ] `uv run pre-commit run --all-files` passes (ruff lint, ruff-format, django-upgrade).
  - `uv run ruff check . --fix` to help resolve linting errors
- [ ] `uv run python -Wd manage.py check` reviewed for new deprecation warnings.
- [ ] `uv run pytest` full test suite passes.
- [ ] `uv run python manage.py makemigrations --dry-run` — if it reports any
  migrations, they belong in this PR and the change should be marked Breaking.
  (Use the plain `--dry-run`, not `--check --dry-run`: with `--check` the command
  exits 1 and prints *nothing*, which reads exactly like "no migrations needed".)
- [ ] Docker builds succeed if touched.
- [ ] Docs updated under `docs/` if behaviour, settings, or APIs changed.
- [ ] `release_notes.md` updated if user-facing.
- [ ] No secrets, debug code, or stray `print`/`console.log` left in.

### Review Code

- [ ] Reviewer verifies the target branch.
- [ ] For non-trivial changes: pulled the branch, ran it locally, and confirmed it works as described.
- [ ] CI is green; test suite and `pre-commit` pass.
- [ ] Reads through all changed files for anything weird or unintended.
- [ ] Logic is correct; edge cases and error handling considered.
- [ ] Migrations are sane and reversible where possible.
- [ ] Changes are consistent with the modernization direction (no reintroduction
      of patterns being migrated away from, e.g. flake8/yapf-era conventions).
- [ ] No security concerns (input validation, permissions, injection, XSS).
- [ ] Comments are sufficient and not overkill.
- [ ] Typos, spellcheck, professional.
- [ ] Reviewer sends back review.

### Edits

- [ ] Coder fixes any issues.
- [ ] Reviewer approves.

> Maintainer review is required for breaking changes, new features, and migrations.
> For bug fixes, refactors, and docs-only changes, reviewer approval is sufficient to merge.

---

## Testing Notes

Run via `uv`, on Linux, against sqlite (file-backed and in-memory).

- Full suite: **1206 passed, 55 skipped**
- `uv run pre-commit run --all-files`: exit 0
- `uv run python -Wd manage.py check`: *"System check identified no issues (0 silenced)."*
- GUI suite exercised in all four combinations — `pytest -m selenium` with
  `SELENIUM_BROWSER` ∈ {firefox, chromium} × `SELENIUM_HEADLESS` ∈ {True, False}
- Screenshot-on-failure verified by forcing a failure and opening the resulting
  PNG — the first version of the hook wrote blank images, because `tearDown`
  navigated to `about:blank` before the report hook ran. Capture and write are
  now separate steps.
- Chromium flakiness measured rather than assumed: repeated A/B runs, not single
  runs, before concluding anything about a change's effect.

**On `makemigrations --dry-run`:** it reports migrations on this branch, but
`develop` reports the identical 208-line operation list (diffed both ways).
This branch adds none — the pending migrations are the project's pre-existing
ones.

**For reviewers:** 6,445 of the ~9,100 added lines are the regenerated schema
SVG. The real change is roughly 2,700 lines. Suggested reading order —
`conftest.py`, `qatrack/qatrack_core/tests/live.py`, the test files, then the docs.

Draft because I'd rather agree on the harness before the tests written against
it multiply.

## Screenshots

N/A — no UI changes.

### CODE REVIEW

**Comment by nsmela** [comment]
## Blocking

**1. `qa/tests/test_selenium.py:884` — DB assert races the POST**
`wait_for_ajax()` returns the moment the new page loads; the assert can run before the service event is committed. Verified 4/12 failures → 0/8 with:
```python
self.wait_until(
    lambda: models.TestListInstance.objects.first().serviceevents_initiated.count() == 1,
    "the service event to be created",
)
```

**2. `qa/tests/test_selenium.py:1083` — clicks a transiently hidden element**
`qabulkreview.js` hides toggle index 0, then columnFilter (`sPlaceHolder: "head:after"`) moves that row to the bottom, shifting it to index 1. In between, index 0 is `display:none` — observed in 8/40 page loads on `ed68777`, always with `jQuery.active == 0`. Gate on initialisation, not on `wait_for_ajax()`:
```python
def initialised(d):
    els = d.find_elements(By.CLASS_NAME, "test-selected-toggle")
    if len(els) < 2:
        return False
    shown = [e.is_displayed() for e in els]
    return els if (not all(shown) and shown[0]) else False

self.wait.until(initialised, "the bulk review JS to hide the filter row's select-all checkbox")[0].click()
```
Note: "click the first *visible* toggle" is **not** sufficient — it's satisfied before `qabulkreview.js` binds its handlers.

**3. `qatrack_core/tests/live.py` — untouched; timeouts stay undiagnosable**
Seven message-less waits (348, 356, 407, 415, 443, 493, and the `find_element` monkeypatch at 196-203). The monkeypatch is the important one: every `find_element` in the suite raises an empty-message `TimeoutException`. This is why the original CI failure couldn't be localised.

## Should fix

**4. `test_selenium.py:879` — waits on nothing**
`$service_areas_fake.change()` (`sl_serviceevent.js:300-309`) fires no AJAX. Dead wait; remove or replace with the real condition.

**5. `settings.py:874` clobbers `local_settings.py`**
`SELENIUM_CHROMIUM_BINARY_PATH = ''` is assigned *after* the `local_settings` import (line 754), so setting it there is silently discarded. Only `local_test_settings.py` works — the docstring should say so explicitly.

**6. Selenium Manager will use an incompatible chromedriver**
With a mismatched `chromedriver` on `PATH`, Selenium Manager prints its own incompatibility warning and then uses it anyway → `SessionNotCreatedException`. The "no manual driver install needed on any host" claim in `settings.py` is too strong; document `SELENIUM_CHROMIUM_DRIVER_PATH` as the fallback.

## Questions for the author

**7. Six skipped stubs were deleted in `ed68777`**
`test_perform_boolean`, `test_perform_upload`, `test_autosave_datetime_roundtrip`, `test_review_reject`, `test_review_non_bulk`, `test_unreview`. These documented real coverage gaps. Intentional, or collateral?

**8. Latent app bug (separate issue, not this PR)**
`qabulkreview.js:9` `.eq(0).hide()` is only correct because it wins the race against `listable`. If `listable` ever initialises first, it permanently hides the header select-all users actually click.

## Context on CI

All 12 checks pass on `ed68777`, including all three Selenium jobs — but items 1 and 2 both still reproduce locally on that exact commit. Green here is one sample of a race, not a fix; don't treat it as evidence either was resolved.

File: deploy/dev/local_test_settings.dev.py
Comment (crcrewso) [inline deploy/dev/local_test_settings.dev.py]: removed, sample for each test deployment type replaces this one file since parallel testing of multiple back ends from one development environment will be supported.

File: docs/developer/images/qatrack_schema_4.0.0.svg
Comment (crcrewso) [inline docs/developer/images/qatrack_schema_4.0.0.svg]: This might have been a mistake, I don't think I meant to include the updated schema in this commit.

File: docs/developer/guide.rst [hunk 1]
```diff
@@ -168,37 +185,93 @@ files from the deploy subdirectory and then create your database:
...
+     - A MySQL/MariaDB deployment (likewise with ``.sql`` setup scripts).
+   * - ``deploy/win/local_settings.py``
+     - A Windows/MS SQL Server deployment.
+
+Copy whichever one matches your target to ``qatrack/local_settings.py`` and
+edit it from there. For full deployment instructions see the
+:doc:`installation guides </install/install>`, not this page.
 
 
 Understanding the Settings Files
```
Comment (crcrewso) [inline docs/developer/guide.rst]: local_test_settings.<engine>.py should probably be covered here

File: qatrack/reports/tests/test_base.py
Comment (crcrewso) [inline qatrack/reports/tests/test_base.py]: moved selenium tests out of base test class
